### PR TITLE
User and team name

### PR DESCRIPTION
**Pull request by danpersa**

### CODE REVIEW

File: src/main/scala/org/zalando/spearheads/innkeeper/api/JsonDtos.scala [hunk 1]
```diff
@@ -22,10 +22,29 @@ object RouteName {
...
+case class UserName(name: String) extends AnyRef {
+
+  def apply(name: Option[String]) = name match {
+    case Some(name) => UserName(name)
+    case _          => throw InvalidUsereNameException
+  }
+}
+
+object UserName {
+  def apply(name: Option[String]): UserName = name match {
```
Comment (mo-gr) [inline src/main/scala/org/zalando/spearheads/innkeeper/api/JsonDtos.scala]: That seems like a weird syntax/API to me. Having an apply with an Option that throws on none... Why not just have an apply with a pure string and let the caller of that thing handle the case of empty Options? This way, I always have to wrap a String into an Option to create a Username? seems weird

File: src/main/scala/org/zalando/spearheads/innkeeper/api/JsonProtocols.scala [hunk 2]
```diff
@@ -165,6 +165,18 @@ object JsonProtocols {
...
 
-  implicit val routeOutFormat = jsonFormat(RouteOut, "id", "name", "route", "created_at", "activate_at", "owned_by_team", "description", "deleted_at")
+  implicit object UserNameFormat extends RootJsonFormat[UserName] {
+
+    override def write(userName: UserName): JsValue = JsString(userName.name)
+
+    override def read(json: JsValue): UserName = {
+      json match {
+        case JsString(value) => UserName(value)
+        case _               => throw new DeserializationException("Error deserializing the user name")
```
Comment (mo-gr) [inline src/main/scala/org/zalando/spearheads/innkeeper/api/JsonProtocols.scala]: probably helpful to contain the original son in the exception?

File: src/main/scala/org/zalando/spearheads/innkeeper/utils/HttpClient.scala [hunk 3]
```diff
@@ -45,12 +45,12 @@ class AkkaHttpClient @Inject() (implicit val actorSystem: ActorSystem,
 
     for {
       jsonString <- Try {
-        logger.debug(s"We call the service with the url: ${uri}")
+        logger.info(s"We call the service with the url: ${uri}")
```
Comment (mo-gr) [inline src/main/scala/org/zalando/spearheads/innkeeper/utils/HttpClient.scala]: that should remain debug... that is not info-level information... that is purely debug-relevant

**Comment by mo-gr** [comment]
apart from my comments, lgfm

File: src/main/scala/org/zalando/spearheads/innkeeper/api/JsonDtos.scala (hunk 1 above)
Comment (danpersa) [inline src/main/scala/org/zalando/spearheads/innkeeper/api/JsonDtos.scala]: This is only an overriden constructor... So you are able to pass both a string or an option.

I removed the `apply` method in the case class, as it's useless.

File: src/main/scala/org/zalando/spearheads/innkeeper/api/JsonProtocols.scala (hunk 2 above)
Comment (danpersa) [inline src/main/scala/org/zalando/spearheads/innkeeper/api/JsonProtocols.scala]: we don't give our clients lots of details... a short message and a 404 is enough for now :)

**Pull request merged**